lib/ecto/adapters/sql.ex

```diff
@@ -612,7 +612,11 @@ defmodule Ecto.Adapters.SQL do
         {:ok, Enum.zip(returning, values)}
 
       {:ok, %{num_rows: 0}} ->
-        if on_conflict == :nothing, do: {:ok, []}, else: {:error, :stale}
+        if on_conflict == :nothing do
+          {:ok, []}
+        else
+          {:error, :no_change}
```

Comment:
I don't think we should change this code. The change in ecto/repo/schema should be enough. :)